apps/gatsby/src/Sidebar.js

```diff
@@ -42,8 +40,72 @@ export default class Sidebar extends React.Component {
     this.debounceInterval = setInterval(this.refreshPreview, 1000);
   };
 
-  componentDidMount() {
+  buildSlug = async () => {
+    const {urlConstructors, previewUrl} = this.sdk.parameters.installation;
+    //Find the url constructor for the given contentType
+    const constructor = urlConstructors ? urlConstructors.find(
+      constructor => constructor.id === this.sdk.contentType.sys.id
+    ) : undefined;
+    // If there is no constructor set the url as the base preview
+    if (!constructor){
+      return
+    }
+
+    //QUESTION: What does the locale object in the sdk look like when there are multiple locales available. Need to account for spaces that have more than one locale, so they can have previews everywhere.
```

Comment:
https://github.com/contentful/ui-extensions-sdk/blob/861d665cb3c79d7e20261bd6b0a55cf2dd903890/lib/types.ts#L380-L394

This should be it